src/averageAges.js

```diff
@@ -1,61 +1,58 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function avarageAge(persons) {
+  const ages = persons.reduce(
+    (sum, person) => sum + (person.died - person.born),
+    0
+  );
+
+  const averageAge = ages / persons.length;
+
+  return averageAge;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century !== undefined
+    ? people.filter(
+      (person) =>
+        Math.ceil(person.died / 100) === century && person.sex === 'm'
+    )
+    : people.filter((person) => person.sex === 'm');
```

Comment:
Mb we don't need to write "person.sex === 'm' " twice and can use one variables instead